sentry/src/main/java/io/sentry/config/Lookup.java

```diff
@@ -1,90 +1,123 @@
+    private static Lookup instance;
+
+    private final ConfigurationProvider highPriorityProvider;
+    private final ConfigurationProvider lowPriorityProvider;
 
     /**
-     * The filename of the Sentry configuration file.
-     */
-    private static final String CONFIG_FILE_NAME = "sentry.properties";
-    /**
-     * Properties loaded from the Sentry configuration file, or null if no file was
-     * found or it failed to parse.
-     */
-    private static Properties configProps;
-    /**
-     * Whether or not to check the JNDI system. Exists so that JNDI checks can be disabled
-     * the first time the class is not found.
+     * Constructs a new instance of Lookup with the default configuration providers.
      */
-    private static boolean checkJndi = true;
-
-    static {
-        String filePath = getConfigFilePath();
-        InputStream input = null;
-        try {
-            input = getInputStream(filePath);
-
-            if (input != null) {
-                configProps = new Properties();
-                configProps.load(input);
-            } else {
-                logger.debug("Sentry configuration file not found in filesystem or classpath: '{}'.", filePath);
-            }
-        } catch (Exception e) {
-            logger.error("Error loading Sentry configuration file '{}': ", filePath, e);
-        } finally {
-            Util.closeQuietly(input);
-        }
+    public Lookup() {
+        this(new CompoundConfigurationProvider(getDefaultHighPriorityConfigurationProviders()),
+                new CompoundConfigurationProvider(getDefaultLowPriorityConfigurationProviders()));
     }
 
     /**
-     * Hidden constructor for static utility class.
+     * Constructs a new Lookup instance.
+     *
+     * The two configuration providers are used before and after an attempt to load the configuration property from
+     * the DSN in the {@link #get(String, Dsn)} method.
+     *
+     * @param highPriorityProvider the configuration provider that is consulted before the check in DSN
+     * @param lowPriorityProvider the configuration provider that is consulted only after the high priority one and
+     *                            the DSN
      */
-    private Lookup() {
+    public Lookup(ConfigurationProvider highPriorityProvider, ConfigurationProvider lowPriorityProvider) {
+        this.highPriorityProvider = highPriorityProvider;
+        this.lowPriorityProvider = lowPriorityProvider;
+    }
 
+    private static List<ConfigurationResourceLocator> getDefaultResourceLocators() {
+        return asList(new SystemPropertiesBasedLocator(), new EnvironmentBasedLocator(), new DefaultLocator());
     }
 
-    private static String getConfigFilePath() {
-        String filePath = System.getProperty("sentry.properties.file");
+    private static List<ConfigurationProvider> getDefaultHighPriorityConfigurationProviders() {
+        boolean jndiPresent = JndiSupport.isAvailable();
```

Comment:
We just have to make sure this call won't block the android UI as per #708.

Reply:
Not sure how to test that atm. All I know is that the behavior is the same as it was with the prior implementation of `Lookup` which did the exact same check as is done in `JndiSupport.isAvailable()` during the first config property lookup. This is exactly the behavior now as well, where we instantiate the inner "deprecated" instance of Lookup on the first config lookup lazily.